lms/djangoapps/email_marketing/signals.py

```diff
@@ -54,41 +57,33 @@ def add_email_marketing_cookies(sender, response=None, user=None,
         if sailthru_content:
             post_parms['cookies'] = {'anonymous_interest': sailthru_content}
 
-    try:
-        sailthru_client = SailthruClient(email_config.sailthru_key, email_config.sailthru_secret)
-        log.info(
-            'Sending to Sailthru the user interest cookie [%s] for user [%s]',
-            post_parms.get('cookies', ''),
-            user.email
-        )
-        time_before_call = datetime.datetime.now()
-
-        sailthru_response = \
-            sailthru_client.api_post("user", post_parms)
+    time_before_call = datetime.datetime.now()
+    sailthru_response = get_email_cookies_via_sailthru.delay(user.email, post_parms)
 
-    except SailthruClientError as exc:
+    try:
+        # synchronous call to celery task with a timeout limit
+        sailthru_response.get(timeout=COOKIE_CREATION_REQUEST_TIME_LIMIT_IN_SECONDS,
+                              propagate=True)
+        cookie = sailthru_response.result
+
+    except TimeoutError as exc:
+        log.error("Timeout error while attempting to obtain cookie from Sailthru: %s", unicode(exc))
+    except SailthruClientError:
         log.error("Exception attempting to obtain cookie from Sailthru: %s", unicode(exc))
 
         return response
 
-    if sailthru_response.is_ok():
-        if 'keys' in sailthru_response.json and 'cookie' in sailthru_response.json['keys']:
-            cookie = sailthru_response.json['keys']['cookie']
-
-            response.set_cookie(
-                'sailthru_hid',
-                cookie,
-                max_age=365 * 24 * 60 * 60,  # set for 1 year
-                domain=settings.SESSION_COOKIE_DOMAIN,
-                path='/',
-            )
-            _log_sailthru_api_call_time(time_before_call)
-        else:
-            log.error("No cookie returned attempting to obtain cookie from Sailthru for %s", user.email)
+    if not cookie:
+        return response
     else:
-        error = sailthru_response.get_error()
-        # generally invalid email address
-        log.info("Error attempting to obtain cookie from Sailthru: %s", error.get_message())
+        response.set_cookie(
+            'sailthru_hid',
+            cookie,
+            max_age=365 * 24 * 60 * 60,  # set for 1 year
+            domain=settings.SESSION_COOKIE_DOMAIN,
+            path='/',
+        )
+        _log_sailthru_api_call_time(time_before_call)
```

Comment:
@simon Keeping it for testing purpose on sandbox. Would remove it before merging.